distributed/security.py

```diff
@@ -306,3 +363,24 @@ def get_listen_args(self, role):
             "ssl_context": self._get_tls_context(tls, ssl.Purpose.CLIENT_AUTH),
             "require_encryption": self.require_encryption,
         }
+
+    # make this class frozen so that the validation in __init__ always applies
+    def __setattr__(self, name, value):
+        raise AttributeError(f"Can't set attribute {name!r}")
+
+    def __delattr__(self, name):
+        raise AttributeError(f"Can't delete attribute {name!r}")
+
+    # __setstate__is needed for un-pickling frozen classes
+    def __setstate__(self, state):
```

Comment:
making Security frozen broke pickle support:

https://github.com/dask/distributed/runs/6863709912?check_suite_focus=true#step:11:182

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Miniconda3\envs\dask-distributed\lib\multiprocessing\spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "C:\Miniconda3\envs\dask-distributed\lib\multiprocessing\spawn.py", line 126, in _main
    self = reduction.pickle.load(from_parent)
  File "d:\a\distributed\distributed\distributed\security.py", line 368, in __setattr__
    raise AttributeError(f"Can't set attribute {name!r}")
AttributeError: Can't set attribute 'require_encryption'
```

maybe making it frozen isn't worth it?

Reply:
I've removed the commits making security.Security frozen from this PR